ui/app/AppLayouts/Profile/views/LanguageView.qml

```diff
@@ -12,42 +14,127 @@ import "../popups"
 import "../stores"
 
 Item {
-    id: languageContainer
-    Layout.fillHeight: true
-    Layout.fillWidth: true
-    clip: true
+    id: root
 
     property LanguageStore languageStore
     property int profileContentWidth
-    property Component languagePopup: LanguageModal {
-        languageStore: languageContainer.languageStore
+
+    QtObject {
+        id: d
+        property int margins: 64
+        property int zOnTop: 100
+
+        function setViewIdleState() {
+            languagePicker.close()
+        }
     }
 
-    Item {
+    z: d.zOnTop
+    Layout.fillHeight: true
+    Layout.fillWidth: true
+    clip: true
+
+    onVisibleChanged: { if(!visible) d.setViewIdleState()}
+
+    Component.onCompleted: { root.languageStore.initializeLanguageModel() }
+
+    Column {
+        z: d.zOnTop
+        width: 560 - 2 * Style.current.padding
+        anchors.margins: d.margins
         anchors.top: parent.top
-        anchors.topMargin: 64
         anchors.bottom: parent.bottom
-        width: profileContentWidth
+        anchors.left: parent.left
+        spacing: 45
 
-        anchors.horizontalCenter: parent.horizontalCenter
+        StatusBaseText {
+            id: title
+            text: qsTr("Language & Currency")
+            font.weight: Font.Bold
+            font.pixelSize: 22
+            color: Theme.palette.directColor1
+            anchors.bottomMargin: Style.current.padding
+        }
 
-        Column {
-            id: generalColumn
+        Item {
+            id: language
             width: parent.width
+            height: 38
+            z: d.zOnTop
 
-            StatusListItem {
-                //% "Language"
-                title: qsTrId("language")
-                label: localAppSettings.locale === "" ? qsTrId("default") : localAppSettings.locale
-                components: [
-                    StatusIcon {
-                        icon: "chevron-down"
-                        rotation: 270
-                        color: Theme.palette.baseColor1
+            StatusBaseText {
+                text: qsTr("Language")
+                anchors.left: parent.left
+                font.pixelSize: 15
+                color: Theme.palette.directColor1
+            }
+            StatusListPicker {
+                id: languagePicker
+
+                property string newKey
+
+                Timer {
```

Comment:
How long is this pause on your machine? Will it grow with an entry-level laptop? How about in the future when the complexity grows? Won't a popup with a message like "Loading the new translation" provide a better future-proof user experience?

Reply:
We can ask @John-44 about the design / loading icon / message. 

Now what it's happening without the pause is that when user selects a new language it takes less than 1 second to execute the action so the UI seems to respond slow by marking the radiobutton this 1 second later. With the pause incorporated the UI is inmediately changed, so no visual block, but the action itself is executed 100 ms later (no perception to the user).

Reply:
Ok! Let's do like this. Here the [#task](https://github.com/status-im/status-desktop/issues/5400) to decide how to proceed!